osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -181,15 +181,18 @@ public void Import(string path)
 
         public void Import(IEnumerable<BeatmapSetInfo> beatmapSets)
         {
-            connection.BeginTransaction();
-
-            foreach (var s in beatmapSets)
+            lock (connection)

```

Comment:
The connection is also used in other places. I don't think this is thread safe at all. Maybe the content of this method should be scheduled to the main thread instead of locking?